src/scheduler/fifo.c

```diff
@@ -1667,10 +1667,8 @@ run_update_resresv(status *policy, int pbs_sd, server_info *sinfo,
 		}
 
 		update_queue_on_run(qinfo, rr, &old_state);
-		if (flags & NO_ALLPART)
-			update_all_nodepart(policy, sinfo, rr, NO_ALLPART);
-		else
-			update_all_nodepart(policy, sinfo, rr, NO_FLAGS);
+
+		sinfo->pset_metadata_stale = 1;
```

Comment:
You are correct, but remember that nodes can be in many node partitions.  Figuring out those node partitions to update would be an expensive operation.  It's an N^4 operation to determine this plus we'd have to allocate memory where we didn't before.  It's a N^2 operation to update everything, but the constants are higher.  It also makes the optimization much harder if we do what you are suggesting.  We'd have to keep around which node partitions which need updating vs just one bit that says they're stale.

Reply:
Ah, okay, ya N^4 is very bad. Would it be too ugly to collect things that need updating while we are trying to determine where to run a job?

Reply:
We can determine what placement set was used to run the job when node searching, but that doesn't really help us.  Since nodes can be in multiple partitions, We'd have to do the N^4 operation to figure it out.  This is why I opted for the N^2 operation to update all placement set metadata.  The constants on the N^2 operation are much higher than the N^4 operation.  Maybe N^4 isn't as bad.  I've never done the exercise to figure it out.